### ISSUE TITLE
Very slow rendering with dynamic data

### ISSUE THREAD

**Issue by arshadkazmi42** [Devlight/InfiniteCycleViewPager#66]
I am populating viewpaper with data from server. It is taking 5-10 minutes to render 6 pages.

**Comment by msr550** [comment]
Yes, I also observed. It is taking more time to render dynamic. Most of the times ANR coming.

Thanks

**Comment by arshadkazmi42** [comment]
any fixes for this yet?

**Comment by remylavergne** [comment]
Same problem here. In my case, my internet request is very quick, but when I set the adapter, something freeze my UI between :

- myLoop.setAdapter(myInfiniteLoopAdapter);
&
- myLoop.notifyDataSetChanged();

The UI is freezing almost 2 minutes, specially on debug mode. I don't have any problem on my physical device (but in powerless devices, the problem occurs).

**Comment by arshadkazmi42** [comment]
I have checked the core code. Its happening due to layout resizing. Its doing some nested operations for resizing the layout. Looks like developer this repo is not maintained any more.

**Comment by remylavergne** [comment]
Yeap this library is dead :) No response from the creator since 2016. We have change for another lib updated on Android Arsenal.

**Comment by arshadkazmi42** [comment]
@remylavergne can you share the library with me. I am looking for some library similar to this one

**Comment by piyushmishra10** [comment]
The reason behind ANR is: there is a bug in the lib.

Ideally, there shouldn't be a left view when we load the data. Sometimes it's not there, in that case, the loading time is very less.

Most of the time you'll see a left view but you will see the 0th position object only and end position view on the left, in that case, it's making a loop and taking hell lot of time to populate the view. Hence, ANR in some cases.

**Comment by arshadkazmi42** [comment]
@piyushmishra10 Can you fix the issue in the code? And send me the fixed version of this?

**Comment by SUBINPTPM** [comment]
any updates?

**Comment by arshadkazmi42** [comment]
@SUBINPTPM This repository is not maintained anymore, I built my own viewpager, not as good as this.
There are alot of other alternatives, you can check those out

**Comment by SUBINPTPM** [comment]
@arshadkazmi42  link, please .

**Comment by arshadkazmi42** [comment]
Havn't added that code in github. You can search for ViewPagers you will get some good options. I will ping you once i add my code on github7

**Comment by SUBINPTPM** [comment]
@arshadkazmi42  ok
